### ISSUE TITLE
Injected 503 response is not available to inner/outer routes.

### ISSUE THREAD

**Issue by jCalamari** [akka/akka-http#1085]
Given route definition as follows:

```
    addTraceHeaderIfMissing {
      logHttpRequest {
        logHttpResponse {
          withRequestTimeoutResponse(handleTimeoutResponse) {
            logHttpResponse {
              Route.seal(routes)
            }
          }
        }
      }
    }
```

Where `logHttpResponse` is implemented with `mapResponse` directive, both `logHttpResponse` calls  log `200 OK` response if service timeout is reached.

Any clues? Am I missing something? Many thanks!

**Comment by jrudolph** [comment]
Yes, that is indeed surprising but a consequence of the implementation which runs the timeout handling outside of routing. The rationale is to do as little as possible when a timeout occurs to prevent further delays. If you want to log timeouts you need to do so manually in your `handleTimeoutResponse` method.

Let's keep this ticket to improve the documentation about that fact (scaladoc, javadoc, reference documentation for the timeout directives, and the timeout page).

**Comment by jCalamari** [comment]
Thanks @jrudolph for explaining. 

> If you want to log timeouts you need to do so manually in your handleTimeoutResponse method.

This means I will end up with two log entries, 200 and 503. There is no way for inner route to be aware of `timeout has happened` event, is it?

**Comment by jrudolph** [comment]
Yes, right, that's a bit of a disadvantage. Maybe we need to add at least some bit of logging in the internals which is the only place which knows how the race between a proper response and the timeout ended up.
